packages/web-components/src/components/ic-radio-group/ic-radio-group.tsx

```diff
@@ -161,6 +168,29 @@ export class RadioGroup {
     this.radioOptions[0].shadowRoot.querySelector("input").tabIndex =
       this.selectedChild > 0 ? -1 : 0;
 
+    let radioHasAdditionalField = false;
+    let i = 0;
+
+    while (
+      this.radioOptions &&
+      i < this.radioOptions.length &&
+      !radioHasAdditionalField
+    ) {
+      radioHasAdditionalField = isSlotUsed(
```

Comment:
As we already know inside each radio option (via private hasAdditionalField flag), would it be cleaner to pass that information back to the radio group somehow (eg emitting an internal event) rather than duplicating the logic to do the detection?

Reply:
Discussed this further with @ad9242 and the agreed approach was to combine the logic in the while loop into the existing if statement